doc/user-guide/dask.rst

```diff
@@ -2,195 +2,151 @@
 
-.. warning::
+    Or open multiple files with py:func:`~xarray.open_mfdataset`::
 
-    :py:func:`~xarray.open_mfdataset` called without ``chunks`` argument will return
-    dask arrays with chunk sizes equal to the individual files. Re-chunking
-    the dataset after creation with ``ds.chunk()`` will lead to an ineffective use of
-    memory and is not recommended.
+        xr.open_mfdataset('my/files/*.nc', parallel=True)
 
-You'll notice that printing a dataset still shows a preview of array values,
-even if they are actually Dask arrays. We can do this quickly with Dask because
-we only need to compute the first few values (typically from the first block).
-To reveal the true nature of an array, print a DataArray:
+    .. tip::
 
-.. ipython:: python
+        When reading in many netCDF files with py:func:`~xarray.open_mfdataset`, using ``engine=h5netcdf`` can
+        be faster than the default which uses the netCDF4 package.
 
-    ds.temperature
+    Saving a larger-than-memory netCDF file::
 
-Once you've manipulated a Dask array, you can still write a dataset too big to
-fit into memory back to disk by using :py:meth:`~xarray.Dataset.to_netcdf` in the
-usual way.
+       delayed_write = ds.to_netcdf("my-big-file.nc", compute=False)
+       delayed_write.compute()
 
-.. ipython:: python
+    .. note::
 
-    ds.to_netcdf("manipulated-example-data.nc")
+        When using Dask’s distributed scheduler to write NETCDF4 files, it may be necessary to set the environment variable ``HDF5_USE_FILE_LOCKING=FALSE`` to avoid competing locks within the HDF5 SWMR file locking scheme. Note that writing netCDF files with Dask’s distributed scheduler is only supported for the netcdf4 backend.
 
-By setting the ``compute`` argument to ``False``, :py:meth:`~xarray.Dataset.to_netcdf`
-will return a ``dask.delayed`` object that can be computed later.
+    See :ref:`io.netcdf` for more details.
 
-.. ipython:: python
+.. tab:: HDF5
 
-    from dask.diagnostics import ProgressBar
+    Open HDF5 files with :py:func:`~xarray.open_dataset`::
 
-    # or distributed.progress when using the distributed scheduler
-    delayed_obj = ds.to_netcdf("manipulated-example-data.nc", compute=False)
-    with ProgressBar():
-        results = delayed_obj.compute()
+        xr.open_dataset("/path/to/my/file.h5", chunks='auto')
 
-.. ipython:: python
-    :suppress:
+    See :ref:`io.hdf5` for more details.
 
-    os.remove("manipulated-example-data.nc")  # Was not opened.
+.. tab:: GeoTIFF
 
-.. note::
+    Open large geoTIFF files with rioxarray::
 
-    When using Dask's distributed scheduler to write NETCDF4 files,
-    it may be necessary to set the environment variable `HDF5_USE_FILE_LOCKING=FALSE`
-    to avoid competing locks within the HDF5 SWMR file locking scheme. Note that
-    writing netCDF files with Dask's distributed scheduler is only supported for
-    the `netcdf4` backend.
+        xds = rioxarray.open_rasterio("my-satellite-image.tif", chunks='auto')
```

Comment:
Xarray has dedicated documentation on IO for different formats. Unless these functions behaves differently when used with dask than one would expect from the above explanation, I suggest either moving these examples to the dedicated IO docs page or leaving them out.

Reply:
You mean this page https://docs.xarray.dev/en/stable/user-guide/io.html, right?

When I went through those sections they seemed more focused on the file format than Dask, so it seemed like adding in Dask-specific snippets or explanations would be overly complicated for people who don't need to use Dask.

My thought with adding tabs to the Dask page is they quickly show which file formats you can use with Dask, a simple example, and then link out to different sections of https://docs.xarray.dev/en/stable/user-guide/io.html.

Definitely defer to what you think is a better fit for the Xarray docs, though!